### PR TITLE
enable amm test in pool-models 

### PR DESCRIPTION
**Pull request by catShaark**
Closes: #XXX

## What is the purpose of the change

Make test doesn't run the amm_test in pool-models, this PR fixes that

## Brief Changelog

- change name of TestCalculateAmountOutAndIn_InverseRelationship in internal/test_helpers
- add test function to run `BalancertestSuite` and `StableSwapTestSuite`

## Testing and Verifying

`Test_StableSwap_CalculateAmountOutAndIn_InverseRelationship` currently fails because of overflow issue in sdk math.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/gamm/pool-models/balancer/amm_test.go [hunk 1]
```diff
@@ -14,6 +15,10 @@ type BalancerTestSuite struct {
 	test_helpers.CfmmCommonTestSuite
 }
 
+func TestBalancerTestSuite(t *testing.T) {
```
Comment (alexanderbez) [inline x/gamm/pool-models/balancer/amm_test.go:18]: hmmm should this exist in `x/gamm/pool-models/balancer/pool_test.go`?

Comment (catShaark) [reply x/gamm/pool-models/balancer/amm_test.go:18]: I'm not sure whether this should be in

Comment (mattverse) [reply x/gamm/pool-models/balancer/amm_test.go:18]: I think it's fair it exists with wherever `BalancerTestSuite` is.
Personally would prefer both being in `pool_test.go`

File: x/gamm/pool-models/stableswap/amm_test.go [hunk 2]
```diff
@@ -15,6 +16,10 @@ type StableSwapTestSuite struct {
 	test_helpers.CfmmCommonTestSuite
 }
 
+func TestStableSwapTestSuite(t *testing.T) {
```
Comment (mattverse) [inline x/gamm/pool-models/stableswap/amm_test.go:19]: ditto here, would prefer this to be in `pool_test.go`

**Review by mattverse — APPROVED** [review]
Looks good to merge once the test suite set up placing is resolved!

File: x/gamm/pool-models/balancer/amm_test.go (hunk 1 above)
Comment (catShaark) [reply x/gamm/pool-models/balancer/amm_test.go:18]: I'll move it pool_test.go

**Comment by catShaark** [comment]
@mattverse, I've made a new PR https://github.com/osmosis-labs/osmosis/pull/2560 for this since this branch has to many conflicts with main, and apply the changes you request

**Comment by mattverse** [comment]
@catShaark  Nice! Closing this PR in replacement of the new PR u've uploaded

**Pull request closed without merging**